apethoud/happy-cards-v2
# Add card form and holiday picker

### What this PR does

Adds a card form with the following fields:

- Recipient Name
- Recipient Email
- Selected Holiday
- Sender Name
- Sender Email

The Selected Holiday field opens a modal that allows the user to pick a holiday from a list. The holiday list can be toggled between the holidays for the current month and the current year. A search input is also provided to filter the table based on the text entered.

### Screenshots

Card form:
<img width="1061" alt="image" src="https://user-images.githubusercontent.com/22208997/175794928-fdc7009a-5dd8-4a8f-a9d1-e21d3e37eac2.png">

Holiday picker modal:
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/22208997/175794936-b01b1077-21f9-4182-9323-de5f6ea6949a.png">

Filled-out form:
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/22208997/175794949-d4b3b503-0f8f-4b18-a087-21a8d8127b13.png">

### Testing Steps

1. Confirm the UI matches the designs provided in the connected issue.

2.
Given: a user has entered text into each of the input fields
When: the "Submit" button is clicked
Then: the data entered should be logged accurately to the console

3.
When: a user clicks "Choose" under the Pick a Holiday section
Then: a modal should appear that shows a list of holidays to choose from

4.
Given: the Pick a Holiday modal is open and the holiday list is displayed
Then: the number of displayed holidays should be accurately shown to the right of the filter buttons

5.
When: a user clicks a gray button in the Pick a Holiday modal's table header (either "This Month" or "This Year", depending on the state)
Then: the corresponding holiday list should be shown and the clicked button should become purple

6.
When: a user enters text into the search field
Then: the table data should be filtered based on the text entered

7.
When: a user clicks on a holiday in the Pick a Holiday modal
Then: the modal should immediately close and the clicked holiday should be shown under the Pick a Holiday section in place of the choose button.

8.
Given: a holiday has been selected from the Pick a Holiday modal
When: the "Edit" button next to the selected holiday is clicked
Then: the Pick a Holiday modal should be opened and allow a user to select a new holiday to replace the one previously selected

Resolves #3
